kvbc/include/kvbc_app_filter/kvbc_app_filter.h

```diff
@@ -228,9 +181,8 @@ class KvbAppFilter {
 
   uint64_t newestTagSpecificPublicEventGroupId() const;
 
-  // Generate event group ids in batches from storage
-  // We do not want to process in memory, all the event group ids generated in a pruning window
-  std::optional<uint64_t> getNextEventGroupId(std::shared_ptr<EventGroupClientState> &eg_state);
+  // Given a tag-specific public (external) event group id, return the corresponding global event group id
```

Comment:
Is `public` correct for this comment? Aren't we passing a tag-specific event group ID?